site/app/controllers/calendar/CalendarController.php

```diff
@@ -27,8 +31,125 @@ class CalendarController extends AbstractController {
      */
     public function viewCalendar(): WebResponse {
         $user = $this->core->getUser();
-        $gradeables_of_user = GradeableUtils::getAllGradeableListFromUserId($this->core, $user);
 
-        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables_of_user));
+        $calendar_messages = [];
+
+        $courses = $this->core->getQueries()->getCourseForUserId($user->getId());
+
+        $gradeables_of_user = GradeableUtils::getAllGradeableListFromUserId($this->core, $user, $courses, $calendar_messages);
+
+        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables_of_user, $courses, $calendar_messages));
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/calendar")
+     * @return WebResponse
+     */
+    public function viewCourseCalendar(): WebResponse {
+        $calendar_messages = [];
+        $user = $this->core->getUser();
+        $courses = $this->core->getQueries()->getCourseForUserId($user->getId());
+        $name = $this->core->getConfig()->getCourse();
+        // Initialize $oneCourse and get the specific course for the course calendar
+        $oneCourse = $courses[0];
+        foreach ($courses as $course) {
+            if ($course->getTitle() == $name) {
+                $oneCourse = [$course];
+                break;
+            }
+        }
+        $gradeables = GradeableUtils::getGradeablesFromUserAndCourse($this->core, $calendar_messages);
+
+        return new WebResponse(CalendarView::class, 'showCalendar', CalendarInfo::loadGradeableCalendarInfo($this->core, $gradeables, $oneCourse, $calendar_messages), true);
+    }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/calendar/newItem", methods={"POST"})
```

Comment:
I personally dislike camel case routes, and they generally go against the rest of the routing scheme used across Submitty.  Perhaps the route `/courses/{_semester}/{_course}/calendar/items/new` or similar would be a good alternative.